autogluon/utils/tabular/features/abstract_feature_generator.py

```diff
@@ -350,18 +359,16 @@ def check_if_nlp_feature(self, X: Series):
 
     def generate_text_features(self, X: Series, feature: str) -> DataFrame:
         X: DataFrame = X.to_frame(name=feature)
-        X[feature + '.char_count'] = [self.char_count(value) for value in X[feature].values]
-        X[feature + '.word_count'] = [self.word_count(value) for value in X[feature].values]
-        X[feature + '.capital_ratio'] = [self.capital_ratio(value) for value in X[feature].values]
-        X[feature + '.lower_ratio'] = [self.lower_ratio(value) for value in X[feature].values]
-        X[feature + '.digit_ratio'] = [self.digit_ratio(value) for value in X[feature].values]
-        X[feature + '.special_ratio'] = [self.special_ratio(value) for value in X[feature].values]
+        X[feature + '.char_count'] = X[feature].apply(self.char_count)
```

Comment:
How does this affect performance? I remember .apply() having bad performance in pandas which is why I generally avoided it for most things (like using zip list comprehensions as a replacement)

Reply:
Hey Nick, 
You are right, pandas.apply performs slightly worse than list comprehension. Check the notebook: https://github.com/razmikmelikbekyan/ML-playground/blob/master/miscellaneous/Performance%20Checks.ipynb
So I will revert them back to list comprehensions.

Reply:
Thanks for checking!